### DISCUSSION TITLE
GUI not functioning

### DISCUSSION THREAD

**Discussion by Icyyfr** [vsharma-git/virtualHOTAS#1] [Q&A] [answered]
I tried using USB tethering by enabling USB debugging and using adb reverse over the port 5000. I could connect to the server by now entering localhost:5000 in my phone browser but I couldn't interact with any of the buttons or sliders (the whole GUI was greyed out).

**Comment by vsharma-git** [comment]
You dont actually have to do anything like that.
When you open run.bat, you should see an IP address which is not localhost. You just need to open that on your phone.
If you'd like you can email me at [mp72ylvbk@mozmail.com](mailto:mp72ylvbk@mozmail.com) and we can set up a call, if you'd like.
After the call, I can summarize what happened and how we resolved it here.

**Reply by Icyyfr** [reply]
I do not have a wifi to connect to at the moment, so devices aren't on the same network, that is why I used USB.. will connecting the laptop to my phone's hotspot help? (With mobile data off)

**Reply by vsharma-git** [reply]
Yeah! Try that. 
Make a hotspot with your phone. Connect your laptop to your phone's hotspot. 
Then run the app on your laptop. The app will give you the IP address to connect to.

**Reply by Icyyfr** [reply]
I will attach screenshots of the problem. When I use localhost:5000, it runs as expected but when I use the IP address the buttons do not work (except the full screen one which does what it's supposed to do)
Both screenshots from firefox browser on my laptop.
![ip](https://github.com/vsharma-git/virtualHOTAS/assets/76117855/4c4ccd72-eea5-48d5-b015-5443e1e23fce)
![local](https://github.com/vsharma-git/virtualHOTAS/assets/76117855/60b7c07b-4592-498a-aa7c-16f12348d5e3)

**Reply by vsharma-git** [reply, marked as answer]
Thanks for the screenshot. That really helped. 

I'm so sorry, I take my original no-internet message back. You do need internet for this to work. But you only need it in the start so that the app can load. 

What is happening is that the app needs to download the library for the sliders and the styles for the buttons from the internet. Once the page has fully loaded, then you should be able to disable the internet.
